File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/ElectricityPhysicalQuantity.cpp
// ElectricityPhysicalQuantity.cpp 
#include "ElectricityPhysicalQuantity.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ElectricityPhysicalQuantity::ElectricityPhysicalQuantity(TiXmlNode* xmlNode)
: CommodityPhysicalQuantityBase(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //physicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* physicalQuantityNode = xmlNode->FirstChildElement("physicalQuantity");

   if(physicalQuantityNode){physicalQuantityIsNull_ = false;}
   else{physicalQuantityIsNull_ = true;}

   if(physicalQuantityNode)
   {
      for(physicalQuantityNode; physicalQuantityNode; physicalQuantityNode = physicalQuantityNode->NextSiblingElement("physicalQuantity")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityNode , address : " << physicalQuantityNode << std::endl;
          #endif
          if(physicalQuantityNode->Attribute("href") || physicalQuantityNode->Attribute("id"))
          {
              if(physicalQuantityNode->Attribute("id"))
              {
                  physicalQuantityIDRef_ = physicalQuantityNode->Attribute("id");
                  physicalQuantity_.push_back(boost::shared_ptr<ElectricityPhysicalDeliveryQuantity>(new ElectricityPhysicalDeliveryQuantity(physicalQuantityNode)));
                  physicalQuantity_.back()->setID(physicalQuantityIDRef_);
                  IDManager::instance().setID(physicalQuantityIDRef_, physicalQuantity_.back());
              }
              else if(physicalQuantityNode->Attribute("href")) { physicalQuantityIDRef_ = physicalQuantityNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { physicalQuantity_.push_back(boost::shared_ptr<ElectricityPhysicalDeliveryQuantity>(new ElectricityPhysicalDeliveryQuantity(physicalQuantityNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityNode , address : " << physicalQuantityNode << std::endl;
       #endif
   }

   //physicalQuantityScheduleNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* physicalQuantityScheduleNode = xmlNode->FirstChildElement("physicalQuantitySchedule");

   if(physicalQuantityScheduleNode){physicalQuantityScheduleIsNull_ = false;}
   else{physicalQuantityScheduleIsNull_ = true;}

   if(physicalQuantityScheduleNode)
   {
      for(physicalQuantityScheduleNode; physicalQuantityScheduleNode; physicalQuantityScheduleNode = physicalQuantityScheduleNode->NextSiblingElement("physicalQuantitySchedule")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityScheduleNode , address : " << physicalQuantityScheduleNode << std::endl;
          #endif
          if(physicalQuantityScheduleNode->Attribute("href") || physicalQuantityScheduleNode->Attribute("id"))
          {
              if(physicalQuantityScheduleNode->Attribute("id"))
              {
                  physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode->Attribute("id");
                  physicalQuantitySchedule_.push_back(boost::shared_ptr<ElectricityPhysicalDeliveryQuantitySchedule>(new ElectricityPhysicalDeliveryQuantitySchedule(physicalQuantityScheduleNode)));
                  physicalQuantitySchedule_.back()->setID(physicalQuantityScheduleIDRef_);
                  IDManager::instance().setID(physicalQuantityScheduleIDRef_, physicalQuantitySchedule_.back());
              }
              else if(physicalQuantityScheduleNode->Attribute("href")) { physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { physicalQuantitySchedule_.push_back(boost::shared_ptr<ElectricityPhysicalDeliveryQuantitySchedule>(new ElectricityPhysicalDeliveryQuantitySchedule(physicalQuantityScheduleNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityScheduleNode , address : " << physicalQuantityScheduleNode << std::endl;
       #endif
   }

   //totalPhysicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* totalPhysicalQuantityNode = xmlNode->FirstChildElement("totalPhysicalQuantity");

   if(totalPhysicalQuantityNode){totalPhysicalQuantityIsNull_ = false;}
   else{totalPhysicalQuantityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- totalPhysicalQuantityNode , address : " << totalPhysicalQuantityNode << std::endl;
   #endif
   if(totalPhysicalQuantityNode)
   {
      if(totalPhysicalQuantityNode->Attribute("href") || totalPhysicalQuantityNode->Attribute("id"))
      {
          if(totalPhysicalQuantityNode->Attribute("id"))
          {
             totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode->Attribute("id");
             totalPhysicalQuantity_ = boost::shared_ptr<UnitQuantity>(new UnitQuantity(totalPhysicalQuantityNode));
             totalPhysicalQuantity_->setID(totalPhysicalQuantityIDRef_);
             IDManager::instance().setID(totalPhysicalQuantityIDRef_,totalPhysicalQuantity_);
          }
          else if(totalPhysicalQuantityNode->Attribute("href")) { totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { totalPhysicalQuantity_ = boost::shared_ptr<UnitQuantity>(new UnitQuantity(totalPhysicalQuantityNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
std::vector<boost::shared_ptr<ElectricityPhysicalDeliveryQuantity>> ElectricityPhysicalQuantity::getPhysicalQuantity()
{
   if(!this->physicalQuantityIsNull_){
        if(physicalQuantityIDRef_ != ""){
             return this->physicalQuantity_;
        }else{
             return this->physicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<ElectricityPhysicalDeliveryQuantity>>();
   }
}
std::vector<boost::shared_ptr<ElectricityPhysicalDeliveryQuantitySchedule>> ElectricityPhysicalQuantity::getPhysicalQuantitySchedule()
{
   if(!this->physicalQuantityScheduleIsNull_){
        if(physicalQuantityScheduleIDRef_ != ""){
             return this->physicalQuantitySchedule_;
        }else{
             return this->physicalQuantitySchedule_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<ElectricityPhysicalDeliveryQuantitySchedule>>();
   }
}
boost::shared_ptr<UnitQuantity> ElectricityPhysicalQuantity::getTotalPhysicalQuantity()
{
   if(!this->totalPhysicalQuantityIsNull_){
        if(totalPhysicalQuantityIDRef_ != ""){
             return boost::shared_static_cast<UnitQuantity>(IDManager::instance().getID(totalPhysicalQuantityIDRef_));
        }else{
             return this->totalPhysicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<UnitQuantity>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/ElectricityPhysicalQuantity.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ElectricityPhysicalQuantity : CommodityPhysicalQuantityBase
    {
        public ElectricityPhysicalQuantity(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNodeList physicalQuantityNodeList = xmlNode.SelectNodes("physicalQuantity");
            
            if (physicalQuantityNodeList != null)
            {
                this.physicalQuantity_ = new List<ElectricityPhysicalDeliveryQuantity>();
                foreach (XmlNode item in physicalQuantityNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            physicalQuantityIDRef_ = item.Attributes["id"].Value;
                            physicalQuantity_.Add(new ElectricityPhysicalDeliveryQuantity(item));
                            IDManager.SetID(physicalQuantityIDRef_, physicalQuantity_[physicalQuantity_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            physicalQuantityIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        physicalQuantity_.Add(new ElectricityPhysicalDeliveryQuantity(item));
                        }
                    }
                    else
                    {
                        physicalQuantity_.Add(new ElectricityPhysicalDeliveryQuantity(item));
                    }
                }
            }
            
        
            XmlNodeList physicalQuantityScheduleNodeList = xmlNode.SelectNodes("physicalQuantitySchedule");
            
            if (physicalQuantityScheduleNodeList != null)
            {
                this.physicalQuantitySchedule_ = new List<ElectricityPhysicalDeliveryQuantitySchedule>();
                foreach (XmlNode item in physicalQuantityScheduleNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            physicalQuantityScheduleIDRef_ = item.Attributes["id"].Value;
                            physicalQuantitySchedule_.Add(new ElectricityPhysicalDeliveryQuantitySchedule(item));
                            IDManager.SetID(physicalQuantityScheduleIDRef_, physicalQuantitySchedule_[physicalQuantitySchedule_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            physicalQuantityScheduleIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        physicalQuantitySchedule_.Add(new ElectricityPhysicalDeliveryQuantitySchedule(item));
                        }
                    }
                    else
                    {
                        physicalQuantitySchedule_.Add(new ElectricityPhysicalDeliveryQuantitySchedule(item));
                    }
                }
            }
            
        
            XmlNode totalPhysicalQuantityNode = xmlNode.SelectSingleNode("totalPhysicalQuantity");
            
            if (totalPhysicalQuantityNode != null)
            {
                if (totalPhysicalQuantityNode.Attributes["href"] != null || totalPhysicalQuantityNode.Attributes["id"] != null) 
                {
                    if (totalPhysicalQuantityNode.Attributes["id"] != null) 
                    {
                        totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode.Attributes["id"].Value;
                        UnitQuantity ob = new UnitQuantity(totalPhysicalQuantityNode);
                        IDManager.SetID(totalPhysicalQuantityIDRef_, ob);
                    }
                    else if (totalPhysicalQuantityNode.Attributes["href"] != null)
                    {
                        totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        totalPhysicalQuantity_ = new UnitQuantity(totalPhysicalQuantityNode);
                    }
                }
                else
                {
                    totalPhysicalQuantity_ = new UnitQuantity(totalPhysicalQuantityNode);
                }
            }
            
        
        }
        
    
        #region PhysicalQuantity_
        private List<ElectricityPhysicalDeliveryQuantity> physicalQuantity_;
        public List<ElectricityPhysicalDeliveryQuantity> PhysicalQuantity_
        {
            get
            {
                if (this.physicalQuantity_ != null)
                {
                    return this.physicalQuantity_; 
                }
                else if (this.physicalQuantityIDRef_ != null)
                {
                    return this.physicalQuantity_; 
                }
                else
                {
                      return this.physicalQuantity_; 
                }
            }
            set
            {
                if (this.physicalQuantity_ != value)
                {
                    this.physicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string physicalQuantityIDRef_ { get; set; }
        #region PhysicalQuantitySchedule_
        private List<ElectricityPhysicalDeliveryQuantitySchedule> physicalQuantitySchedule_;
        public List<ElectricityPhysicalDeliveryQuantitySchedule> PhysicalQuantitySchedule_
        {
            get
            {
                if (this.physicalQuantitySchedule_ != null)
                {
                    return this.physicalQuantitySchedule_; 
                }
                else if (this.physicalQuantityScheduleIDRef_ != null)
                {
                    return this.physicalQuantitySchedule_; 
                }
                else
                {
                      return this.physicalQuantitySchedule_; 
                }
            }
            set
            {
                if (this.physicalQuantitySchedule_ != value)
                {
                    this.physicalQuantitySchedule_ = value;
                }
            }
        }
        #endregion
        
        public string physicalQuantityScheduleIDRef_ { get; set; }
        #region TotalPhysicalQuantity_
        private UnitQuantity totalPhysicalQuantity_;
        public UnitQuantity TotalPhysicalQuantity_
        {
            get
            {
                if (this.totalPhysicalQuantity_ != null)
                {
                    return this.totalPhysicalQuantity_; 
                }
                else if (this.totalPhysicalQuantityIDRef_ != null)
                {
                    totalPhysicalQuantity_ = IDManager.getID(totalPhysicalQuantityIDRef_) as UnitQuantity;
                    return this.totalPhysicalQuantity_; 
                }
                else
                {
                      return this.totalPhysicalQuantity_; 
                }
            }
            set
            {
                if (this.totalPhysicalQuantity_ != value)
                {
                    this.totalPhysicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string totalPhysicalQuantityIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

